src/pages/ClerkScreen.tsx

```diff
@@ -39,15 +56,77 @@ const ClerkScreen = ({
   toggleLiveMode,
   unconfigure,
 }: Props) => {
-  const changeAppPrecinctId: EventTargetFunction = (event) => {
-    const currentTarget = event.currentTarget as HTMLInputElement
-    const appPrecinctId = currentTarget.value
-    setAppPrecinctId(appPrecinctId)
+  const changeAppPrecinctId: SelectChangeEventFunction = (event) => {
+    setAppPrecinctId(event.currentTarget.value)
   }
 
   const [isTestDeck, setIsTestDeck] = useState(false)
   const showTestDeck = () => setIsTestDeck(true)
   const hideTestDeck = () => setIsTestDeck(false)
+
+  const [isSavingDate, setIsSavingDate] = useState(false)
+  const [isSystemDateModalActive, setIsSystemDateModalActive] = useState(false)
+  const [systemDate, setSystemDate] = useState(new Date())
+  const [systemMeridian, setSystemMeridan] = useState<Meridian>(
+    systemDate.getHours() < 12 ? 'AM' : 'PM'
+  )
+  const [systemTimezoneLabel, setSystemTimezoneLabel] = useState(
```

Comment:
Why store the label? Just store the IANA value and look up the label.

Reply:
Good refactor. Done.